src/parser.ts

```diff
@@ -126,6 +126,22 @@ export function parse(input: string, isStrict = true, dialect: Dialect = 'generi
         cteState.state = tokenState;
         prevState = tokenState;
         continue;
+        // If we're scanning in a CTE, handle someone putting a semicolon anywhere (after 'with',
+        // after semicolon, etc.) along it to "early terminate".
+      } else if (cteState.isCte && cteState.parens === 0 && token.type === 'semicolon') {
```

Comment:
Why the need for `cteState.parens === 0` here? Is it valid to have a semicolon within parens in a cte?

Reply:
Good point. Removed that condition, and now any semicolon will be considered a terminator. Can revisit this in the future if someone complains about it.